### PR TITLE
VIP per namespace for EVH in VCF

### PR DESCRIPTION
**Pull request by monotosh-avi**
In VCF, we would create VIP per namespace, adding support for that in this PR.
We would look for an environment variable USE_VCF to enable this feature.

In this case, a mapping between namespace and hostname is saved to find out if
a hostname is present in multiple namespaces.

The parent VS name is modified in this case to include namespace instead of shard.

To do in next PRs
- UTs, Helm chart modification

### CODE REVIEW

**Comment by monotosh-avi** [comment]
build ako

**Comment by monotosh-avi** [comment]
Error in running tests: /dev/stdout: resource temporarily unavailable

File: internal/nodes/avi_model_evh_nodes.go [hunk 1]
```diff
@@ -1402,7 +1413,6 @@ func (o *AviObjectGraph) DeletePoolForHostnameForEvh(vsName, hostname string, ro
 		// Replace the ingress map for this host.
 		keyToRemove := namespace + "/" + ingName
 		delete(ingressHostMap.HostNameMap, keyToRemove)
-		SharedHostNameLister().Save(hostname, ingressHostMap)
```
Comment (chauhanshubham) [inline internal/nodes/avi_model_evh_nodes.go:1405]: How come this is not required anymore?

Comment (monotosh-avi) [reply internal/nodes/avi_model_evh_nodes.go:1405]: Good catch, deleted by mistake, udpated

**Comment by monotosh-avi** [comment]
error : Observed a panic: runtime.boundsError{x:8, y:8, signed:true, code:0x0} (runtime error: index out of range [8] with length 8)

**Comment by monotosh-avi** [comment]
build ako

File: internal/lib/lib.go [hunk 2]
```diff
@@ -1319,3 +1319,11 @@ func GetK8sMinSupportedVersion() string {
 func GetK8sMaxSupportedVersion() string {
 	return k8sMaxVersion
 }
+
+func IsVCFCluster() bool {
+	vcfCluster := os.Getenv(VCF_CLUSTER)
```
Comment (akshayhavile) [inline internal/lib/lib.go:1324]: How will this value populated to AKO POD? will by default in VCF each node have this?

File: internal/lib/lib.go [hunk 3]
```diff
@@ -1319,3 +1319,11 @@ func GetK8sMinSupportedVersion() string {
...
+	if vcfCluster == "true" {
```
Comment (akshayhavile) [inline internal/lib/lib.go:1325]: we can optimise it as --> return vcfCluster == "true" .

**Review by akshayhavile — APPROVED** [review]

**Review by chauhanshubham — APPROVED** [review]

**Pull request merged**